MekHQ/src/mekhq/gui/CampaignGUI.java

```diff
@@ -2591,43 +2557,61 @@ private void refreshPartsAvailability() {
     private ActionScheduler fundsScheduler = new ActionScheduler(this::refreshFunds);
 
     @Subscribe
-    public void handleDayEnding(DayEndingEvent ev) {
+    public void handleDayEnding(DayEndingEvent evt) {
         // first check for overdue loan payments - don't allow advancement until
         // these are addressed
         if (getCampaign().checkOverDueLoans()) {
             refreshFunds();
             showOverdueLoansDialog();
-            ev.cancel();
+            evt.cancel();
             return;
         }
+
         if (getCampaign().checkRetirementDefections()) {
             showRetirementDefectionDialog();
-            ev.cancel();
+            evt.cancel();
             return;
         }
+
         if (getCampaign().checkYearlyRetirements()) {
             showRetirementDefectionDialog();
-            ev.cancel();
+            evt.cancel();
+            return;
+        }
+
+        if (new UnmaintainedUnitsNagDialog(getFrame(), getCampaign()).showDialog().isCancelled()) {
```

Comment:
Random thought: a helper class with static methods could clean these up (I'd then throw them all in there as static classes):
```java
if (!Nags.checkForUnmaintainedUnits(getFrame(), getCampaign())) {
```

Reply:
I'm not the biggest fan of statics, and... it would be another class call for a single line, which is rarely worth it.